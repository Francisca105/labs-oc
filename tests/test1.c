#include "../code/4.1/L1Cache.h"

// Checks if the program is safe from out of bounds memory acess

int main() {

  resetTime();
  initCache();

  int i = 0, value;
  write(i, (unsigned char *)(&i));
  printf("Write; Address %d; Value %d; Time %d\n", i, i, getTime());

  read(i, (unsigned char *)(&value));
  printf("Read; Address %d; Value %d; Time %d\n", i, value, getTime());

  i = DRAM_SIZE - 1;
  write(i, (unsigned char *)(&i));
  printf("Write; Address %d; Value %d; Time %d\n", i, i, getTime());

  read(i, (unsigned char *)(&value));
  printf("Read; Address %d; Value %d; Time %d\n", i, value, getTime());

  i = DRAM_SIZE;
  write(i, (unsigned char *)(&i));
  printf("Write; Address %d; Value %d; Time %d\n", i, i, getTime());

  read(i, (unsigned char *)(&value));
  printf("Read; Address %d; Value %d; Time %d\n", i, value, getTime());
  
  return 0;
}