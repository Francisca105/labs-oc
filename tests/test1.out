Write; Address 0; Value 0; Time 101
Read; Address 0; Value 0; Time 102
Write; Address 65535; Value 65535; Time 203
Read; Address 65535; Value 65535; Time 204
